cmd/auth_proxy/main.go

```diff
@@ -0,0 +1,344 @@
+	var authSubjects []auth.SubjectsWithFilters
+
+	if len(config.AuthPolicies) > 0 {
+		if err := json.Unmarshal([]byte(config.AuthPolicies), &authSubjects); err != nil {
+			return nil, fmt.Errorf("failed to parse policies: %w", err)
+		}
+	}
+
+	handler := &ProxyHandler{
+		kubeClient:   kubeclient.Get(ctx),
+		authVerifier: auth.NewVerifier(ctx, nil, nil, configMapWatcher),
+		config:       config,
+		authSubjects: authSubjects,
+	}
+
+	handler.withContext = func(ctx context.Context) context.Context {
+		return logging.WithLogger(featureStore.ToContext(ctx), logger)
+	}
+
+	httpProxy, err := httpReverseProxy(config)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create HTTP proxy: %w", err)
+	}
+
+	httpsProxy, err := httpsReverseProxy(config)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create HTTPS proxy: %w", err)
+	}
+
+	handler.httpProxy = httpProxy
+	handler.httpsProxy = httpsProxy
+
+	return handler, nil
+}
+
+// createServerManager creates the TLS-enabled server manager
+func createServerManager(ctx context.Context, config envConfig, handler *ProxyHandler, logger *zap.SugaredLogger, configMapWatcher *configmap.InformedWatcher) (*eventingtls.ServerManager, error) {
+	var tlsConfig *tls.Config
+	if handler.config.SinkTLSCertPath != nil && handler.config.SinkTLSKeyPath != nil {
+		var err error
+		tlsConfig, err = getServerTLSConfig(*handler.config.SinkTLSCertPath, *handler.config.SinkTLSKeyPath)
+		if err != nil {
+			return nil, fmt.Errorf("failed to get TLS config: %w", err)
+		}
+		logger.Info("TLS config loaded successfully")
+	}
+
+	serverManager, err := eventingtls.NewServerManager(ctx,
+		kncloudevents.NewHTTPEventReceiver(config.ProxyHTTPPort),
+		kncloudevents.NewHTTPEventReceiver(config.ProxyHTTPSPort,
+			kncloudevents.WithTLSConfig(tlsConfig)),
+		handler,
+		configMapWatcher,
+	)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create server manager: %w", err)
+	}
+
+	return serverManager, nil
+}
+
+// startServices starts all background services (configmap watcher and informers)
+func startServices(ctx context.Context, informers []controller.Informer, configMapWatcher *configmap.InformedWatcher, logger *zap.SugaredLogger) error {
+	logger.Debug("Starting ConfigMap watcher")
+	if err := configMapWatcher.Start(ctx.Done()); err != nil {
+		return fmt.Errorf("failed to start ConfigMap watcher: %w", err)
+	}
+
+	logger.Info("Starting informers")
+	if err := controller.StartInformers(ctx.Done(), informers...); err != nil {
+		return fmt.Errorf("failed to start informers: %w", err)
+	}
+
+	return nil
+}
+
+func (h *ProxyHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	ctx := h.withContext(r.Context())
+	logger := logging.FromContext(ctx)
+	features := feature.FromContext(ctx)
+
+	logger.Debugf("Handling request to %s", r.RequestURI)
+
+	err := h.authVerifier.VerifyRequestFromSubjectsWithFilters(ctx, features, h.config.SinkAudience, h.authSubjects, h.config.SinkNamespace, r, w)
```

Comment:
@creydr I may be missing something but since we aren't copying the request here would this have the same problem for structured events that #8710 is trying to avoid?